utils/frame/benchmarking-cli/src/machine/hw.rs

```diff
@@ -0,0 +1,191 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains types to define hardware requirements.
+
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use std::fmt;
+
+lazy_static! {
+	/// The hardware requirements as measured on reference hardware.
+	///
+	/// These values are provided by Parity, however it is possible
+	/// to use your own requirements if you are running a custom chain.
+	///
+	/// The reference hardware is describe here:
+	/// <https://wiki.polkadot.network/docs/maintain-guides-how-to-validate-polkadot>
+	pub static ref POLKADOT_REFERENCE_HARDWARE: HwRequirements = {
+		let raw = include_bytes!("polkadot_reference_hardware.json").as_slice();
+		serde_json::from_slice(raw).expect("Hardcoded data is known good; qed")
+	};
+}
+
+/// Multiple requirements for the hardware.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct HwRequirements(pub Vec<HwRequirement>);
```

Comment:
totally opinionated: not a fan of mixing acronyms and non-acronyms. 

Given that this file is called `hw.rs` (and should have been called `hardware.rs`), I would just drop all the `Hw` prefix from the types. The `Hw` is already in the context. In other files, you can use these types and traits as `hw::Requirements` to make it more explicit.

Reply:
I was not sure how this is normally handled, thanks for the advice. I changed it :+1: